pkg/kruize/getPerformanceProfileMetrics.go

```diff
@@ -0,0 +1,39 @@
+package kruize
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/stolostron/recommends/pkg/utils"
+
+	promv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
+
+	klog "k8s.io/klog/v2"
+)
+
+/* Given a prometheus query returns the reult as float64 */
+func getResults(query string) float64 {
+	var value float64
+	//setup context with a timeout to avoid blocking
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+	defer cancel()
+
+	v1api := promv1.NewAPI(utils.PromClient)
+	res, _, err := v1api.Query(ctx, query, time.Now())
+	if err != nil {
+		if errors.Is(err, context.DeadlineExceeded) {
+			klog.Errorf("API query timed out: %v", err)
+		}
+		klog.Errorf("API query failed: %v", err)
+	}
+
+	vector := res.(model.Vector)
+
+	for _, sample := range vector {
+		value = (float64)(sample.Value)
+	}
```

Comment:
`value` is always assigned the last value in the vector.  Is this what we want?  If yes, then you don't need the for loop, you can just access the last item.

```go
value := (float64)(vector[len(vector) -1].Value)
```

Reply:
@jlpadilla This did not work . So skipped it.

Reply:
ok, can you add a TODO comment to revisit later?